README.md

```diff
@@ -1,3 +1,26 @@
 # @outmark/anno-type
 
 A twoslash-inspired developer-oriented type-checking solution.
+
+## Getting Started
+
+As of now, this is still a very early prototype with little to no documentation and unstable if not faulty code.
+
+You may get some idea on how the tool works by checking out the package scripts in the test folder:
+
+- To run all the examples at once:
+
+  ```sh
+  # You can also just run this script in the test folder directly
+  yarn workspace anno-type-test annotate:examples
+  ```
+
+- To run a specific example with more verbose output:
+  
+  ```sh
+  # Adding 1 to 3 exclamation marks anywhere dumps the current state.
+  # In this example the !! comes at the end and so it will dump the final state.
+  yarn workspace anno-type-test annotate:twoslash:examples \!\!
+  ```
+
+  > **Note**: Interactive shells often require escaping each `\!` in a command line input.
```

Comment:
```suggestion
  > **Note**: Interactive shells often require escaping each `\!` in a command line input.

- To run `annotate-twoslash` commands with a specified `workspace`:

  \```sh
  # Resolving paths ahead of time is the current design choice.
  # This ensures that platform specificity is in tact without additional dependencies.
  yarn workspace anno-type-test annotate-twoslash -al -o=output/temp $(realpath test/sources/examples/*.[tj]s)
  \```

  > **Note**: Wildcards are handled by the interactive shell or by the package manager for package scripts, while the workspace package names and paths are always handled by the package manager itself.

- To run `annotate-twoslash` commands with *package-root*-relative paths:

  \```sh
  # …
  yarn annotate-twoslash -al -o=test/output/temp test/sources/examples/*.[tj]s
  \```

  > **Note**: Bare paths are always resolved relative to the package root and will throw if this cannot be resolved.

- To run `annotate-twoslash` commands with *working-directory*-relative paths:

  \```sh
  # …
  yarn annotate-twoslash -al -o=t../output/temp ./examples/*.[tj]s
  \```

  > **Note**: Interactive shells commonly retain the `.` or `..` in a command line input.
```

A **Path Resolution** section can also be include for clarifying the design choices and how they are intended to mitigate ambiguities along the following lines:

> 1. An implicit or explicit `root-path` is used to simplify the resolution semantics with the following rules:
>    1. any *bare-specifier* is expected to be *package-root*-relative.
>    2. any *relative-specifier* is expected to be *working-directory*-relative.
>    3. any *absolute-specifier* is expected to be nested relative to an implicit or explicit `--root-path`.
> 2. An implicit `root-path` will default to:
>    1. the resolved *package-root* for a given *bare-specifier* or *package-scoped-absolute-specifier*.
>    2. the *specifier-relative-root* for a given *relative-specifier*.
>    3. the *parent-directory* for a *non-package-scoped-absolute-specifier*.
> 3. The *package-root* is always resolved once for each unique specifier using the [`require.resolve`](https://nodejs.org/dist/latest/docs/api/modules.html#requireresolverequest-options) irrespective of *package-scope*.
>    > **Note**: Opting to alter the default resolution behaviour may require consideration.

Those would then clarify what changes need to take place to iron out the obscurities in the implementation to make it stable enough to use.